spec/tbd_tests_spec.rb

```diff
@@ -2718,4 +2718,100 @@
       end
     end
   end
+
+  it "can generate multiple KIVA exposed perimeters (midrise apts - variant)" do
+    translator = OpenStudio::OSVersion::VersionTranslator.new
+    path = OpenStudio::Path.new(File.dirname(__FILE__) + "/files/midrise_KIVA.osm")
+    os_model = translator.loadModel(path)
+    expect(os_model.empty?).to be(false)
+    os_model = os_model.get
+
+    # Reset all ground-facing surfaces as "foundations".
+    os_model.getSurfaces.each do |s|
+      next unless s.outsideBoundaryCondition.downcase == "ground"
+      s.setOutsideBoundaryCondition("Foundation")
```

Comment:
I would suggest not setting this here because it puts the model in a temporarily invalid state (because the boundary condition is "Foundation" but there is no foundation object assigned). The boundary condition will be set to "Foundation" when you call `setAdjacentFoundation` and pass in a foundation object below:
https://github.com/NREL/OpenStudio/blob/develop/src/model/Surface.cpp#L1919

Not setting the model to invalid states was always a challenge because often you have to move through invalid states when modifying a model.  It's probably ok to do things as you are because the model is valid at the beginning and end of the measure.  I would recommend against having the user set the boundary condition to "Foundation" in the app before running the measure because that saves the model in the invalid state, which results in EnergyPlus crashing as you found out.

Reply:
Of course, I'm only setting this up for a small RSpec test - it's not part of the measure per se (so there's no real risk). BTW, not linking a surface with a foundation object does not render the model invalid (EnergyPlus will generate a default foundation object if missing). The issue is with the exposed perimeter (which EnergyPlus will not generate).

I understand the reservations, but the simple fact is that users can currently pick "Foundation" (either in the OpenStudio App or SDK) and therefore generate an invalid model. For the Kiva option to work, users have to select "Foundation" for targeted floors (and optionally walls). Sloped terrain and walkout basements are classic cases where some walls will be tagged as foundation walls wile others will instead be outside-facing. There is no way to make an educated guess, apart from tagging some surfaces with "Foundation". At least the method renders an invalid OSM into a valid one.

Reply:
The --force checkbox/argument should accomplish what you suggest. Requires more formal RSpec tests (to do).